model/qti/event/UpdatedItemEventDispatcher.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ */
+
+namespace oat\taoQtiItem\model\qti\event;
+
+use core_kernel_classes_Resource;
+use oat\oatbox\event\EventManagerAwareTrait;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoItems\model\event\ItemUpdatedEvent;
+use oat\taoQtiItem\model\qti\Item;
+use oat\taoQtiItem\model\qti\parser\IncludedElementIdsExtractor;
+
+class UpdatedItemEventDispatcher extends ConfigurableService
+{
+    use EventManagerAwareTrait;
+
+    public function dispatch(Item $qtiItem, core_kernel_classes_Resource $rdfItem): void
+    {
+        $this->getEventManager()->trigger(
+            new ItemUpdatedEvent(
+                $rdfItem->getUri(),
+                [
+                    'includedElementIds' => $this->getIncludedElementIdsExtractor()->extract($qtiItem)
```

Comment:
I would put this key as public const

Reply:
also is it possible to put more verbose key, smth like media, or at least include instead of included?

Reply:
I kept as a private const (only used here, so no need for external usage). I changed to `includeElementIds` cause the ids belong to a include in the item, which could be many things and the client can decide what to do.

Reply:
<xi:nclude> can only be a sharedstimulus, e.q. any xml. If we have only one media, it is not using xinlude at all. Its why Im asking if we have to take care about these medias

Reply:
I see. Now we only care about shared stimulus, but other media will be handled in other task, so we will probably have another parameter `mediaIds`. I would prefer keep then separately in the event payload cause we can have different actions with them.